ui/app/components/app/connected-sites-list/connected-sites-list.component.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import Button from '../../ui/button'
+
+export default class ConnectedSitesList extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    renderableDomains: PropTypes.arrayOf(PropTypes.shape({
+      name: PropTypes.string,
+      icon: PropTypes.string,
+      key: PropTypes.string,
+      lastConnectedTime: PropTypes.string,
+      permissionDescriptions: PropTypes.array,
+    })).isRequired,
+    domains: PropTypes.object.isRequired,
+    showDisconnectAccountModal: PropTypes.func.isRequired,
+    showDisconnectAllModal: PropTypes.func.isRequired,
+  }
+
+  state = {
+    expandedDomain: '',
+    iconError: '',
+  }
+
+  handleDomainItemClick (domainKey) {
+    if (this.state.expandedDomain === domainKey) {
+      this.setState({ expandedDomain: '' })
+    } else {
+      this.setState({ expandedDomain: domainKey })
+    }
+  }
+
+  render () {
+    const { renderableDomains, domains, showDisconnectAccountModal, showDisconnectAllModal } = this.props
+    const { expandedDomain } = this.state
+    const { t } = this.context
+
+    return (
+      <div className="connected-sites-list">
+        {
+          renderableDomains.map((domain, domainIndex) => {
+            const domainIsExpanded = expandedDomain === domain.key
+            return (
+              <div
+                className={classnames('connected-sites-list__domain', {
+                  'connected-sites-list__domain--expanded': domainIsExpanded,
+                })}
+                key={`connected-domain-${domainIndex}`}
+              >
+                <div className="connected-sites-list__domain-item" onClick={ () => this.handleDomainItemClick(domain.key) }>
+                  <div className="connected-sites-list__domain-item-info-container">
+                    <div className="connected-sites-list__identicon-container">
```

Comment:
Made this comment above on another instance of this: ` <div className="connected-sites-list__identicon-container">` can become a reusable component

Reply:
bbc8d9a9c